### PR TITLE
Fixed type incompatibility for size prop of AppButton in OfferActions Component

### LINKED ISSUE #2979: (SP: 1) Fix type incompatibility for size prop of AppButton in OfferActions Component
There is a type mismatch error in the OfferActions component where the `size` prop passed to `AppButton` is incompatible with the expected type in `AppButtonProps`.

File Path:
`src/components/offer-card/offer-actions/OfferActions.tsx`

Code:
image

Problem:
image

### PR DESCRIPTION
**Pull request by SofiiaYevush**
Extended the `size` prop type for the `AppButton` component to include the `SizeEnum` values (`'small' | 'medium' | 'large' | 'extraLarge' | 'xxl' `). This enables passing `size` values as strings, improving compatibility with MUI.

This change is aligned with SizeEnum definitions:
`common.enums.ts`:
image

### CODE REVIEW

**Review by markgol777 — APPROVED** [review]

File: src/components/app-button/AppButton.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ interface AppButtonProps extends Omit<ButtonProps, 'size'> {
   disabled?: boolean
   component?: ElementType
   to?: string
-  size?: SizeEnum | null
+  size?: 'small' | 'medium' | 'large' | 'extraLarge' | 'xxl' | SizeEnum | null
```
Comment (ShadowOfTheSpace) [inline src/components/app-button/AppButton.tsx]: You can do it in a cleaner way
```suggestion
  size?: `${SizeEnum}` | SizeEnum | null
```

**Review by ShadowOfTheSpace — CHANGES_REQUESTED** [review]

File: src/components/app-button/AppButton.tsx (hunk 1 above)
Comment (ShadowOfTheSpace) [reply src/components/app-button/AppButton.tsx]: And also, why did you estimate this issue for 2 SP? Did you have some problems finding out what caused this error?

Comment (SofiiaYevush) [reply src/components/app-button/AppButton.tsx]: > And also, why did you estimate this issue for 2 SP? Did you have some problems finding out what caused this error?

I tried several other ways to fix the error so as not to change the `AppButton` on which many other files depended, but none of them were good. So I defined 2 SP as it took a while to find the best solution that would not cause errors in other files. Do you suggest estimating less?

Comment (SofiiaYevush) [reply src/components/app-button/AppButton.tsx]: > You can do it in a cleaner way

Done.

Comment (ShadowOfTheSpace) [reply src/components/app-button/AppButton.tsx]: > > And also, why did you estimate this issue for 2 SP? Did you have some problems finding out what caused this error?
> 
> I tried several other ways to fix the error so as not to change the `AppButton` on which many other files depended, but none of them were good. So I defined 2 SP as it took a while to find the best solution that would not cause errors in other files. Do you suggest estimating less?

Actually, estimation of the bug should be done by you, and only you. But in my opinion, 2 SP is a little bit too much for a one line fix. Especially when we will replace AppButton with Button from design-system folder, so this fix is temporary. But, as I said before - it's up to you.

**Review by ShadowOfTheSpace — APPROVED** [review]

**Review by luiqor — APPROVED** [review]

**Pull request merged**